### ISSUE TITLE
Character set defaults and overrides

### ISSUE THREAD

**Issue by dgobbi** [dgobbi/vtk-dicom#108]
If SpecificCharacterSet is missing, it would be useful to assign a default character set to use when interpreting a data set.  Likewise, if SpecificCharacterSet is set but is incorrect, it would be nice to be able to override it.

Since the character set is assigned to each value while the data is being parsed, it seems that the parser would be be best place to perform the override.

**Comment by dgobbi** [comment]
It's worthwhile to discuss specific use cases.

If SpecificCharacterSet is not set, its most likely that the stored characters are (a) ASCII, (b) ISO--8895-1, (c) Windows-1252, where each of these is backwards compatible with the preceding and Windows-1252 is not listed in the DICOM standard.

I have also seen EUC-KR (ISO 2022 IR 149 without escape codes) without SpecificCharacterSet.  Undoubtedly, there are old DICOM files from Taiwan that use Big5, though I haven't seen them yet. 

Also not yet seen are data sets from Russia, the Baltic states, Central Europe, and Arabia where the most common character sets are not the ISO ones supported by DICOM.

In summary, to completely support this "default character set" feature, I would have to decode character sets that aren't supported by the DICOM standard.  Also see !103.

Another thorny issue is how to edit such data sets.  When written back to disk, the character set could be converted to one that is supported by DICOM, or could be left as-is.

**Comment by dgobbi** [comment]
Small addition Re: Chinese character sets.  Agfa conformance statement lists these non-standard mappings for SpecificCharacterSet:

    "ISO 2022 GBK" -> GB 2313-80  (a typo'd alias for GB2312)
    "ISO 2022 B5" -> Big 5

The first of these might be worth adding, for compatibility.  I wonder if there is a semi-algorithmic mapping from Big 5 to GB18030 that I could use for the other?

**Comment by dgobbi** [comment]
With regards to what defaults should be allowed, the list for HL7 can be used as a guideline.

**Comment by dgobbi** [comment]
Fixed as of ef8e2805 (but without any additional character sets).
